pkgs/development/node-packages/overrides.nix

```diff
@@ -23,6 +23,13 @@ final: prev: {
     prePatch = ''
       export NG_CLI_ANALYTICS=false
     '';
+    nativeBuildInputs = [ pkgs.installShellFiles ];
+    postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+    for shell in bash zsh; do
```

Comment:
You can avoid the temporary file by using the different invocation for `installShellCompletion`:
```
      installShellCompletion --cmd ng \
        --bash <($out/bin/ng completion script) \
        --zsh <($out/bin/ng completion script)
```

I don't know anything about zsh, does it indeed use the same completions as bash?

Reply:
https://angular.dev/cli/completion/script
According to the docs the script should be compatible with both